docs/src/prototypes/dropdowns/utils.ts

```diff
@@ -0,0 +1,36 @@
+export const removeElement = (element: string | HTMLElement): HTMLElement => {
+  const elementToRemove = typeof element === 'string' ? document.getElementById(element) : element
+  return elementToRemove.parentNode.removeChild(elementToRemove)
+}
+
+export const insertNodeAtCursorPosition = (params: { id?: string; text?: string } = {}) => {
+  const { id, text } = params
+  if (!id && !text) {
+    throw '[insertNodeAtCursorPosition]: at least one parameter has to be supplied'
+  }
+
+  if (!window.getSelection) {
+    return
+  }
+
+  const sel = window.getSelection()
+  if (!sel.getRangeAt || !sel.rangeCount) {
+    return
+  }
+
+  const range = sel.getRangeAt(0)
+
+  if (text && !id) {
+    const textNode = document.createTextNode(text)
+    range.insertNode(textNode)
+    range.setStartAfter(textNode)
+    return
+  }
+
+  const elem = document.createElement('span')
+  elem.id = id
+  if (text) {
+    elem.innerText = text
+  }
+  range.insertNode(elem)
+}
```

Comment:
https://github.com/stardust-ui/react/pull/901#discussion_r256902962 from @layershifter 

Is there any third party dependency that does this work? That is fully tested and supported?
Why we can't use [`caret-pos`](https://www.npmjs.com/package/caret-pos)?

Reply:
@layershifter Just wanted to keep things minimal and avoid  introducing extra dependencies just for a prototype, clients using this code will end up needing it too.